### ISSUE TITLE
compatibility with esbuild

### ISSUE THREAD

**Issue by alexandervandekleutab** [lukeautry/tsoa#1069]
When importing stuff like `Route`, `Body`, etc from `tsoa`, bundling code with esbuild fails and tries to use `tsoa` as the cli.

## Sorting
  - [x ] bug report
  - [ ] feature request
  - [ ] support request

- I confirm that I
  - [ x] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Steps to Reproduce

- install `esbuild` and `tsoa`
- copy the tsoa example from the docs
- bundle it with `esbuild src/server.ts target=node --outfile=index.js`
- try running it with `node index.js`
- instead of starting the server as expected, the bundled code is treated as if it were the command line tsoa function.

This can be solved by importing from `@tsoa/runtime` directly.

Version of the library: `"tsoa": "^3.9.0"`
Version of NodeJS: `v14.17.0`

- Confirm you were using yarn not npm:  [x]

**Comment by WoH** [comment]
If this is not a misconfiguration, I'd suspect this would more likely be an esbuild issue.

**Comment by WoH** [comment]
Guess I was wrong, seems like ts-node has similar issues: https://github.com/lukeautry/tsoa/issues/1090
Marking this as a bug.

**Comment by jpmtrabbold** [comment]
Also, info from [here](https://github.com/evanw/esbuild/issues/560):

> nodejs v15 deprecated usage of `module.parent` to detect if module is loaded or running as standalone:
> `[DEP0144] DeprecationWarning: module.parent is deprecated due to accuracy issues. Please use require.main to find program entry point instead.`

From [node's documentation](https://nodejs.org/api/all.html#modules_accessing_the_main_module):

> "When a file is run directly from Node.js, `require.main` is set to its module. That means that it is possible to determine whether a file has been run directly by testing `require.main === module`.
> 
> For a file `foo.js`, this will be true if run via `node foo.js`, but false if run by `require('./foo')`."

So - would you guys be okay to try to use `require.main === module` instead of `!module.parent`?

**Comment by WoH** [comment]
> Also, info from [here](https://github.com/evanw/esbuild/issues/560):
> 
> > nodejs v15 deprecated usage of `module.parent` to detect if module is loaded or running as standalone:
> > `[DEP0144] DeprecationWarning: module.parent is deprecated due to accuracy issues. Please use require.main to find program entry point instead.`
> 
> From [node's documentation](https://nodejs.org/api/all.html#modules_accessing_the_main_module):
> 
> > "When a file is run directly from Node.js, `require.main` is set to its module. That means that it is possible to determine whether a file has been run directly by testing `require.main === module`.
> > 
> > For a file `foo.js`, this will be true if run via `node foo.js`, but false if run by `require('./foo')`."
> 
> So - would you guys be okay to try to use `require.main === module` instead of `!module.parent`?
> 
> 
> 

TL;Dr: Yes, feel free to send a PR.

It's a better/recommended way of checking anyway and has no compatibility impact. Not sure why esbuild seems to strip that out (?).

**Comment by jpmtrabbold** [comment]
@WoH great! PR done, tests passed. The PR is minimal, but let me know if you have any problems with it. Thanks
